### PR TITLE
Fix several depth buffer issues.

### LINKED ISSUE #12229: Sequence ZRamp does not behave correctly for sprites that specify frame offsets
Railway tracks don't render properly in the editor sidebar or on the map.  This is probably related to depth metadata.

More specifically, the Z coordinate for sprites assumes that the sprite is centered at the requested position. We need to account for the extra `ZRamp * Offset.Y` term.

### LINKED ISSUE #19577: TS refinery Zoffset

### PR DESCRIPTION
**Pull request by pchote**
This PR fixes a few issues with our depth buffer rendering, focusing on making `Offset` + `ZRamp: 1` work consistently and intuitively. The comments and commit messages should hopefully explain things well if you go commit by commit, and the improved depth preview can be used to verify the before and after behaviour.

Depends on #19559.

Fixes #12229.
Fixes #19577.
Supersedes #19551.

### CODE REVIEW

**Review by chrisforbes — DISMISSED** [review]

File: OpenRA.Game/Traits/World/DebugVisualizations.cs [hunk 1]
```diff
@@ -20,7 +20,49 @@ public class DebugVisualizations
 		public bool CombatGeometry;
 		public bool RenderGeometry;
 		public bool ScreenMap;
-		public bool DepthBuffer;
 		public bool ActorTags;
+
+		bool depthBufferDirty = true; // Force normal rendering at the start of a new world
```
Comment (abcdefg30) [inline OpenRA.Game/Traits/World/DebugVisualizations.cs]:
```suggestion
		// Force normal rendering at the start of a new world
		bool depthBufferDirty = true;
```

Comment (pchote) [reply OpenRA.Game/Traits/World/DebugVisualizations.cs]: Done, and also improved the wording of the comment.

File: mods/ts/rules/gdi-structures.yaml [hunk 2]
```diff
@@ -262,9 +262,6 @@ GAWEAP:
 	WithIdleOverlay@TURBINES:
 		RequiresCondition: !build-incomplete
 		Sequence: idle-turbines
-	WithIdleOverlay@BIB:
-		RequiresCondition: !build-incomplete
-		Sequence: bib
```
Comment (Mailaender) [inline mods/ts/rules/gdi-structures.yaml:269]: Is this an accident? It doesn't seem to get replaced with anything.

Comment (pchote) [reply mods/ts/rules/gdi-structures.yaml:269]: The sequences were set up incorrectly. In the original game `gaweapbb.shp` is used with `gaweap.shp` when the building is idle, and both sprites are swapped out with `gaweap_1.shp` and `gaweap_2.shp` when the door opens.

We are using `gaweap_1.shp` and `gaweap_2.shp` when idle too, so the `gaweapbb.shp` was actually drawing a second redundant bib over the top of the normal bib.

**Comment by dnqbob** [comment]
Is #19438 related to this fix?

**Comment by teinarss** [comment]
Needs a rebase

**Comment by pchote** [comment]
Rebased.

**Comment by teinarss** [comment]
Looks like Mammoth MK2 still have some problems under bridges.
image

**Comment by pchote** [comment]
That's a different issue with the unit renderering and not something that can be fixed here:


See that the bridge is consistent with the surrounding terrain, but that the mammoth tank is poking through. The original game applied [specific hacks](https://modenc.renegadeprojects.com/ZFudgeBridge) to tall units to prevent this (and its not clear yet how we're going to be able to replicate it...).

**Comment by pchote** [comment]
The same thing happens in the original game if we you delete the rules.ini ZFudge override:

**Review by teinarss — DISMISSED** [review]

**Comment by reaperrr** [comment]
Haven't had time to test this yet, but does this possibly fix #19167 as well?

**Comment by Smittytron** [comment]
> Haven't had time to test this yet, but does this possibly fix #19167 as well?

Tested and banshees still disappear.

**Comment by reaperrr** [comment]
The map importer doesn't seem to be subcell-aware, there's 2 soldiers on the same subcell on the right of the GDI APC in the neutral city base.

👍 after fixing the position of one of them.

**Comment by pchote** [comment]
Updated by fixing the map importer instead.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/f349522a05d4e734bde0ec1f1db96c77bc7982eb...60d159286bc5da014ea9be63497f78ceab283d92

**Pull request merged**